### PR TITLE
Upgrade Solidity compiler to 0.5.3

### PR DESCRIPTION
**Pull request by Krigpl**
And remove all errors from the contracts. All of them look nonfunctional, so nothing basically changed.

### CODE REVIEW

File: contracts/GNTDeposit.sol [hunk 1]
```diff
@@ -137,7 +137,7 @@ contract GNTDeposit is ReceivingContract, Ownable {
         emit Lock(msg.sender);
     }
 
-    function onTokenReceived(address _from, uint256 _amount, bytes /* _data */) public onlyToken {
+    function onTokenReceived(address _from, uint256 _amount, bytes memory /* _data */) public onlyToken {
```
Comment (chfast) [inline contracts/GNTDeposit.sol]: The the `memory` exactly means? How does it work when receiving an external call?

File: contracts/GNTPaymentChannels.sol [hunk 2]
```diff
@@ -68,13 +68,13 @@ contract GNTPaymentChannels is ReceivingContract {
         pure
         returns (bool)
     {
-        return _owner == ecrecover(keccak256("\x19Ethereum Signed Message:\n72", _owner, _receiver, _value), _v, _r, _s);
+        return _owner == ecrecover(keccak256(abi.encodePacked("\x19Ethereum Signed Message:\n72", _owner, _receiver, _value)), _v, _r, _s);
```
Comment (chfast) [inline contracts/GNTPaymentChannels.sol]: Is this exactly the same as was before?

Comment (Krigpl) [reply contracts/GNTPaymentChannels.sol]: Yup, even compiler suggests using exactly that to achieve v0.4 compatibility. Also tests still pass.

File: contracts/GNTDeposit.sol (hunk 1 above)
Comment (Krigpl) [reply contracts/GNTDeposit.sol]: This `memory` here is just for clarity, it's not like you can put anything else here. For `public` methods it has to be `memory` while for `external` it's `calldata`. And that means that for an external here the calldata will be copied to memory and cost some gas.
At least that's my understanding.

Comment (chfast) [reply contracts/GNTDeposit.sol]: Why this function is not `external` in the first place?
If the `data` param is not used, can you make it `calldata`?

**Pull request merged**